composer.json

```diff
@@ -26,15 +28,29 @@
     },
     "sort-packages": true
   },
+  "extra": {
```

Comment:
Why install this way? Why not just install via composer https://packagist.org/packages/phpstan/phpstan and use `--ignore-platform-reqs` for composer? See https://getcomposer.org/doc/03-cli.md#install-i

Reply:
That would require you to use `ignore-platform-reqs` _every single time_ though. It's not only annoying, but also dangerous.

Should we ever add dependency XYZ with PHP requirement >= 8 without noticing that requirement, we would not catch it because we ignore platform reqs. And then potentially cause breakage.

Also, this is only a dev dependency, so something like civicrm/composer-downloads-plugin is way easier to handle this.

It's battle tested in the https://github.com/ampproject/amp-wp repo, FWIW.

Reply:
I just feel like the is a misuse of composer when you can download via composer. 

Does this download cache on build as it is not included in the same way other packages?

Reply:
Yes, the download system uses composer's built-in cache.